geode-redis/src/main/java/org/apache/geode/redis/internal/Subscriptions.java

```diff
@@ -55,30 +55,37 @@ public boolean exists(Object channelOrPattern, Client client) {
    * @param channelOrPattern the channel or pattern
    * @return a list of subscriptions
    */
-  public List<Subscription> findSubscriptions(String channelOrPattern) {
-    return this.subscriptions.stream()
+  public synchronized List<Subscription> findSubscriptions(String channelOrPattern) {
+    return subscriptions.stream()
         .filter(subscription -> subscription.matches(channelOrPattern))
         .collect(Collectors.toList());
   }
 
   /**
    * Add a new subscription
    */
-  public void add(Subscription subscription) {
-    this.subscriptions.add(subscription);
+  public synchronized void add(Subscription subscription) {
+    subscriptions.add(subscription);
   }
 
   /**
    * Remove all subscriptions for a given client
    */
-  public void remove(Client client) {
-    this.subscriptions.removeIf(subscription -> subscription.matchesClient(client));
+  public synchronized void remove(Client client) {
+    subscriptions.removeIf(subscription -> subscription.matchesClient(client));
   }
 
   /**
    * Remove a single subscription
    */
-  public void remove(Object channel, Client client) {
-    this.subscriptions.removeIf(subscription -> subscription.isEqualTo(channel, client));
+  public synchronized void remove(Object channel, Client client) {
+    subscriptions.removeIf(subscription -> subscription.isEqualTo(channel, client));
+  }
+
+  /**
+   * @return the total number of all local subscriptions
+   */
+  public int size() {
```

Comment:
Technically I think size() needs to be synchronized as well.

Reply:
I've switched to using a `CopyOnWriteArrayList`.